expo/expo
# [dev-menu] Allow registerDevMenuItems in non-dev bundles

# Why

Related: https://github.com/expo/eas-cli/pull/2214

This is the alternative fix to https://github.com/expo/eas-cli/pull/2050 (which is being reverted in PR above).

# How

I think this will Just Work? Need to test.

I read through the pr that added it and didn't see anything that indicated why this would only work in dev bundles: https://github.com/expo/expo/pull/17528

# Test Plan

(have not completed this yet, but the theoretical flow is:)
1. create app
2. add dev client
3. add JS that calls `registerDevMenuItems` in bundle
4. publish to eas update
5. load update in dev client, ensure it works

Edit: I can't seem to get this set up to test with `et gba`. Here's what I tried:
```
et gba -n testwat45 expo-file-system expo-font expo-modules-core expo-dev-menu expo-dev-client expo-dev-launcher expo-updates expo-updates-interface
eas init
eas update:configure
eas update
npx expo prebuild
cd ios
pod install
// open in xcode, build, launch published update
// see that react isn't working (error with call to `useContext`??)
```

# Checklist
